content/en/blog/_posts/2025-04-15-wi-for-image-pulls.md

```diff
@@ -0,0 +1,169 @@
+
+Currently, Kubernetes administrators have two primary options
+for handling private container image pulls:
+
+1. **Image pull secrets stored in the Kubernetes API**
+   - These secrets are long-lived, making them hard to rotate.
+   - They must be explicitly attached to a service account or pod.
+   - Compromise of a pull secret can lead to unauthorized image access.
+
+2. **Kubelet credential providers**
+   - These providers fetch credentials dynamically at the node level.
+   - Any pod running on the node can access the same credentials.
+   - There’s no per-workload isolation, increasing security risks.
+
+Neither approach aligns with the principles of **least privilege**
+or **ephemeral authentication**, leaving Kubernetes with a security gap.
+
+## The Solution: Service Account Token Integration for Kubelet Credential Providers
+
+This new enhancement enables kubelet credential providers
+to use **workload identity** when fetching image registry credentials.
+Instead of relying on long-lived secrets, credential providers can use
+service account tokens to request short-lived credentials
+tied to a specific pod’s identity.
+
+This approach provides:
+
+- **Workload-specific authentication**:
+  Image pull credentials are scoped to a particular workload.
+- **Ephemeral credentials**:
+  Tokens are automatically rotated, eliminating the risks of long-lived secrets.
+- **Seamless integration**:
+  Works with existing Kubernetes authentication mechanisms,
+  aligning with cloud-native security best practices.
+
+## How it works
+
+### 1. Service Account tokens for credential providers
+
+Kubelet generates **short-lived, automatically rotated** tokens for service accounts
+if the credential provider it communicates with has opted into receiving
+a service account token for image pulls.
+These tokens conform to OIDC ID token semantics
+and are provided to the credential provider
+as part of the `CredentialProviderRequest`.
+The credential provider can then use this token
+to authenticate with an external service.
+
+### 2. Image registry authentication flow
+
+- When a pod starts, the kubelet requests credentials from a **credential provider**.
+- If the credential provider has opted in,
+  the kubelet generates a **service account token** for the pod.
+- The **service account token is included in the `CredentialProviderRequest`**,
+  allowing the credential provider to authenticate
+  and exchange it for **temporary image pull credentials**
+  from a registry (e.g., AWS ECR, GCP Artifact Registry, Azure ACR).
+- The kubelet then uses these credentials
+  to pull images on behalf of the pod.
+
+### 3. Avoiding long-lived secrets
+
+Unlike traditional image pull secrets, this approach:
+
+- Eliminates the need to store secrets in the Kubernetes API.
+- Ensures credentials are ephemeral and automatically rotated.
+- Provides isolation between workloads, reducing security risks.
+
+## Benefits of this approach
```

Comment:
These two sections look like they could be merged. From the first-time reader perspective, I've just read about some benefits of the approach, and now I am going to read a chapter called "Benefits of this approach"